platform/server/http/Proxy/Proxy.ts

```diff
@@ -0,0 +1,84 @@
+import { Server } from 'net';
+import { IncomingMessage, OutgoingMessage } from 'http';
+import { EventEmitter } from 'events';
+
+import { Logger } from '../../../logging';
+
+export class Proxy extends EventEmitter {
+  private server: Server;
+  private readonly eventHandlers: any;
+  private port: number;
+
+  constructor(private readonly log: Logger, readonly kbnServer: any) {
+    super();
+
+    const forwardEvent = (name: string, ...args: any[]) => {
+      this.log.info(`Event is being forwarded: ${name}`);
+      this.emit(name, ...args);
+    };
+
+    this.eventHandlers = {
+      listening: forwardEvent.bind(this, 'listening'),
+      error: forwardEvent.bind(this, 'error'),
+      clientError: forwardEvent.bind(this, 'clientError'),
+      connection: forwardEvent.bind(this, 'connection')
```

Comment:
Why are these needed? (Might be obvious, just add a comment about it in the code)

Reply:
> Why are these needed?

Hapi relies on these events, some are critical, some aren't:
* `listening`: https://github.com/hapijs/hapi/blob/v14.2.0/lib/connection.js#L120
* `error`: https://github.com/hapijs/hapi/blob/v14.2.0/lib/connection.js#L166
* `clientError`: https://github.com/hapijs/hapi/blob/v14.2.0/lib/connection.js#L92
* `connection`: https://github.com/hapijs/hapi/blob/v14.2.0/lib/connection.js#L142

> Might be obvious, just add a comment about it in the code

Sure